core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java

```diff
@@ -215,7 +215,7 @@ public boolean positionNotDistinctFromRow(int leftBlockIndex, int leftPosition,
             Block leftBlock = channels.get(hashChannel).get(leftBlockIndex);
             Block rightBlock = page.getBlock(rightChannels[i]);
             BlockPositionIsDistinctFrom isDistinctFromOperator = isDistinctFromOperators.get(i);
-            if (!isDistinctFromOperator.isDistinctFrom(leftBlock, leftPosition, rightBlock, rightPosition)) {
+            if (isDistinctFromOperator.isDistinctFrom(leftBlock, leftPosition, rightBlock, rightPosition)) {
```

Comment:
please add a test to `io.trino.operator.TestSimplePagesHashStrategy`
It's not dead code yet (although it should be)
```
        // if compilation fails, use interpreter
        return new SimplePagesHashStrategy(
```

I think we should remove that fallback